### ISSUE TITLE
erlang 23 no crypto:hmac_hash_algorithm

### ISSUE THREAD

**Issue by MessageMap** [brigadier/ewt#1]
Running on erlang 23, the function crypto:hmac_hash_algorithm has been removed

Values passed in to:
sign(?TYPE, Alg, Payload, Key)
Type = 'EWT' , Alg = sha256 , Payload = <<"g3QAAAACZAADYWxnZAAGc2hhMjU2ZAADdHlwZAADRVdU.g3QAAAAEZAADZXhwbgUAJNg02Q5kAARuYW1lbQAAAAVhZG1pbmQABXJvbGVzbAAAAAFtAAAABUFkbWluamQABHVzZXJtAAAABWFkbWlu">> , Key = "efhv+Ttx28Rq46f" 
=CRASH REPORT==== 26-Nov-2020::19:53:40.722891 ===
  crasher:
    initial call: cowboy_stream_h:request_process/3
    pid: <0.1530.0>
    registered_name: []
    exception error: undefined function crypto:hmac_hash_algorithm/0
      in function  ewt:sign/4 (/home/ethan/src/name/_build/default/lib/ewt/src/ewt.erl, line 64)
      in call from ewt:token/4 (/home/ethan/src/name/_build/default/lib/ewt/src/ewt.erl, line 20)

**Comment by brigadier** [comment]
The library doesn't call to `hmac_hash_algorithm` directly, it is used `crypto:hmac/3` which is deprecated in the OTP 23 but still should work for a while. so the error you got looks quite strange unless it is an artefact of the new optimization pass in the 23.

Anyway I think I fixed it in this commit https://github.com/brigadier/ewt/commit/e39656ace1f17efec907e6ee86142020b39f9b64 
Check if it works for you.

upd: yes, I've looked up your fork of ewt and there are differences, including call to `crypto:hmac_hash_algorithm`.
